subsys/bootloader/provision/provision_flash.c

```diff
@@ -45,9 +55,16 @@ int public_key_data_read(u32_t key_idx, u8_t *p_buf, size_t buf_size)
 		return -EINVAL;
 	}
 
-	for (size_t i = 0; i < CONFIG_SB_PUBLIC_KEY_HASH_LEN; i++) {
-		p_buf[i] = p_provision_data->pkd[(key_idx *
-				CONFIG_SB_PUBLIC_KEY_HASH_LEN) + i];
+	p_key = (u32_t *)&p_provision_data->pkd[key_idx *
+		CONFIG_SB_PUBLIC_KEY_HASH_LEN];
+
+	/*
+	 * Ensure word alignment, as provision data might be stored in area
+	 * with word sized read limitation.
+	 */
+	__ASSERT(!(p_key & 3), "Key address is not multiple of 4");
```

Comment:
Could this be done with a static assert?

Reply:
Can use `BUILD_ASSERT` and `BUILD_ASSERT_MSG` for static asserts.

Reply:
I don't think this can be evaluated build time as p_key depends on the argument `key_idx` to deduce its index.

Reply:
You're right, but we can check it indirectly by checking the alignment of `p_provision_data->pkd` and of `CONFIG_SB_PUBLIC_KEY_HASH_LEN`.

I guess this could be in addition to the `__ASSERT()`, but it would be nice since `__ASSERT()`s aren't always executed.

Reply:
I'm not able to use the static assert for anything useful.

`BUILD_ASSERT(NRF_UICR_S->OTP);`
Works, and we have set 

```
static const provision_flash_t *p_provision_data =
	(provision_flash_t *)NRF_UICR_S->OTP;
```

But, `BUILD_ASSERT(p_provision_data->pkd);`  does not work. So look like run time assert is the only thing we get.

Reply:
I guess it should be

    static const provision_flash_t * const p_provision_data = ...

But that probably doesn't change much. You could static assert on `offsetof(provision_flash_t, pkd)`

Reply:
Should be fixed now, please check it.